src/main/java/gregtech/common/metatileentities/multi/multiblockpart/appeng/MetaTileEntityMEInputHatch.java

```diff
@@ -269,4 +272,57 @@ public MultiblockAbility<IFluidTank> getAbility() {
     public void registerAbilities(List<IFluidTank> list) {
         list.addAll(Arrays.asList(this.getAEFluidHandler().getInventory()));
     }
+
+    @Override
+    public final void onDataStickLeftClick(EntityPlayer player, ItemStack dataStick) {
+        NBTTagCompound tag = new NBTTagCompound();
+        tag.setTag("MEInputHatch", writeConfigToTag());
+        dataStick.setTagCompound(tag);
+        dataStick.setTranslatableName("gregtech.machine.me.fluid_import.data_stick.name");
+        player.sendStatusMessage(new TextComponentTranslation("gregtech.machine.me.import_copy_settings"), true);
+    }
+
+    protected NBTTagCompound writeConfigToTag() {
+        NBTTagCompound tag = new NBTTagCompound();
+        NBTTagCompound configStacks = new NBTTagCompound();
+        tag.setTag("ConfigStacks", configStacks);
+        for (int i = 0; i < CONFIG_SIZE; i++) {
+            var slot = this.aeFluidHandler.getInventory()[i];
+            IAEFluidStack config = slot.getConfig();
+            if (config == null) {
+                continue;
+            }
+            NBTTagCompound stackNbt = new NBTTagCompound();
+            config.writeToNBT(stackNbt);
+            configStacks.setTag(Integer.toString(i), stackNbt);
+        }
+        return tag;
+    }
+
+    @Override
+    public final boolean onDataStickRightClick(EntityPlayer player, ItemStack dataStick) {
+        NBTTagCompound tag = dataStick.getTagCompound();
+        if (tag == null || !tag.hasKey("MEInputHatch")) {
+            return false;
+        }
+        readConfigFromTag(tag.getCompoundTag("MEInputHatch"));
+        syncME();
+        player.sendStatusMessage(new TextComponentTranslation("gregtech.machine.me.import_paste_settings"), true);
+        return true;
+    }
+
+    protected void readConfigFromTag(NBTTagCompound tag) {
+        if (tag.hasKey("ConfigStacks")) {
+            NBTTagCompound configStacks = tag.getCompoundTag("ConfigStacks");
+            for (int i = 0; i < CONFIG_SIZE; i++) {
```

Comment:
I think we should still do it this way since when you paste the data stick, if the stick doesn't have a configured item/fluid for that slot then it still needs to ensure that the slot's old configuration is cleared